allennlp/nn/util.py

```diff
@@ -1561,8 +1561,8 @@ def add_sentence_boundary_token_ids(
     elif len(tensor_shape) == 3:
         tensor_with_boundary_tokens[:, 1:-1, :] = tensor
         for i, j in enumerate(sequence_lengths):
-            tensor_with_boundary_tokens[i, 0, :] = sentence_begin_token
-            tensor_with_boundary_tokens[i, j + 1, :] = sentence_end_token
+            tensor_with_boundary_tokens[i, 0, :] = sentence_begin_token.detach().to(tensor.device)
+            tensor_with_boundary_tokens[i, j + 1, :] = sentence_end_token.detach().to(tensor.device)
```

Comment:
Why is this necessary?

Reply:
@tpanza, this looks good by and large, but I'd like to understand what this change is for. If we're just setting a scalar in an existing tensor, we don't have to move devices for it explicitly, do we? Is there a computation graph attached to `sentence_begin_token` and `sentence_end_token` that needs to be detached?